File: seed/php-sdk/exhaustive/no-custom-config/src/Endpoints/Pagination/Requests/ListItemsRequest.php
<?php

namespace Seed\Endpoints\Pagination\Requests;

use Seed\Core\Json\JsonSerializableType;

class ListItemsRequest extends JsonSerializableType
{
    /**
     * @var ?string $cursor The cursor for pagination
     */
    public ?string $cursor;

    /**
     * @var ?int $limit Maximum number of items to return
     */
    public ?int $limit;

    /**
     * @param array{
     *   cursor?: ?string,
     *   limit?: ?int,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->cursor = $values['cursor'] ?? null;
        $this->limit = $values['limit'] ?? null;
    }
}

File: seed/csharp-sdk/exhaustive/redact-response-body-on-error/src/SeedExhaustive/Endpoints/Pagination/Requests/ListItemsRequest.cs
using System.Text.Json.Serialization;
using SeedExhaustive.Core;

namespace SeedExhaustive.Endpoints;

[Serializable]
public record ListItemsRequest
{
    /// <summary>
    /// The cursor for pagination
    /// </summary>
    [JsonIgnore]
    public string? Cursor { get; set; }

    /// <summary>
    /// Maximum number of items to return
    /// </summary>
    [JsonIgnore]
    public int? Limit { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
